src/dataframerow/dataframerow.jl

```diff
@@ -36,3 +36,36 @@ Base.next(r::DataFrameRow, s) = ((_names(r)[s], r[s]), s + 1)
 Base.done(r::DataFrameRow, s) = s > length(r)
 
 Base.convert(::Type{Array}, r::DataFrameRow) = convert(Array, r.df[r.row,:])
+
+function Base.hash(r::DataFrameRow, h::UInt)
+    for col in columns(r.df)
+        if isna(col, r.row)
+            h = hash(false, h)
+        else
+            h = hash(true,hash(col[r.row], h))
+        end
+    end
+    return h
+end
+
+# compare two elements in the array
+_isequal( a::Array, i::Int, j::Int ) = isequal(a[i], a[j])
+
+# compare two elements in the data array
+function _isequal( a::DataArray, i::Int, j::Int )
+    if isna(a, i) return isna(a, j)
+    else return !isna(a, j) && isequal(a.data[i], a.data[j]) end
+end
+
+# compare two elements in the pooled array
+# NOTE assume there are no duplicated elements in the pool
+_isequal( a::PooledDataArray, i::Int, j::Int ) = isequal(a.refs[i], a.refs[j])
+
+function Base.isequal(r1::DataFrameRow, r2::DataFrameRow)
+    if r1.df !== r2.df throw( ArgumentError("Comparing rows from different frames not supported") ) end
+    if r1.row == r2.row return true end
+    for col in columns(r1.df)
+        if !_isequal(col, r1.row, r2.row) return false end
+    end
```

Comment:
Separate lines for `if`, `else`, `cond`, `end`, etc. -- I think basic style is outlined in CONTRIBUTING.md.